test/js/bun/test/mock-test.test.ts

```diff
@@ -0,0 +1,39 @@
+import { test, mock, expect } from "bun:test";
```

Comment:
looks good, would add test cases for:
* async fn that resolves
* async fn that rejects
* fn that returns already resolved promise
* fn called with multiple args
* fn that calls another mock fn